### PR TITLE
[QHC-532] Serialize with YAML

### PR DESCRIPTION
**Pull request by fedonman**

### CODE REVIEW

**Comment by jordivallsq** [comment]
why the class Operation in qililab\src\qililab\qprogram\operations\operation.py is the only one not included in the yaml serialization?

**Comment by jordivallsq** [comment]
Great job Vyron, I added some comments and questions and once those are solved we can merge

**Comment by fedonman** [comment]
> why the class Operation in qililab\src\qililab\qprogram\operations\operation.py is the only one not included in the yaml serialization?

Abstract classes don't need to be registered.

File: src/qililab/qprogram/element.py [hunk 1]
```diff
@@ -15,11 +15,12 @@
...
 from uuid import UUID, uuid4
 
-from qililab.utils import DictSerializable
+from qililab.yaml import yaml
 
 
+@yaml.register_class
 @dataclass(frozen=True)
-class Element(DictSerializable):
+class Element:
```
[the remark this answers is not in the document]
Comment (fedonman) [reply src/qililab/qprogram/element.py:23]: This is the old serialization with `qp.to_dict()` that we used to send jobs with API. We have talked with the cloud team to change that as well, and they will use the `serialize(qp)` with the new yaml.

File: src/qililab/qprogram/variable.py [hunk 2]
```diff
@@ -13,29 +13,57 @@
...
-    Frequency = (2,)
-    Phase = (3,)
+    Scalar = 0
+    Time = 1
+    Frequency = 2
+    Phase = 3
     Voltage = 4
 
+    @classmethod
+    def to_yaml(cls, representer, node):
```
[the remark this answers is not in the document]
Comment (fedonman) [reply src/qililab/qprogram/variable.py:33]: Can't be done, this is required by the library.

File: src/qililab/waveforms/waveform.py [hunk 3]
```diff
@@ -17,12 +17,8 @@
 
 import numpy as np
 
-from qililab.utils import DictSerializable
-from qililab.yaml import yaml
 
-
-@yaml.register_class
-class Waveform(DictSerializable):  # pylint: disable=too-few-public-methods, disable=missing-class-docstring
+class Waveform:  # pylint: disable=too-few-public-methods, disable=missing-class-docstring
```
[the remark this answers is not in the document]
Comment (fedonman) [reply src/qililab/waveforms/waveform.py:21]: Same reason as for QProgram above.

**Review by jordivallsq — APPROVED** [review]
All comments replied. Great work!

**Pull request merged**